subprocess/subtree_oom.go

```diff
@@ -0,0 +1,64 @@
+package subprocess
+
+import (
+	"context"
+	"os/exec"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+type OOMTracker struct {
+	WasOOMKilled bool  `json:"was_oom_killed"`
+	IsSudo       bool  `json:"is_sudo"`
+	Pids         []int `json:"pids"`
+}
+
+func NewOOMTracker() *OOMTracker {
+	return &OOMTracker{}
+}
+
+func isSudo(ctx context.Context) (bool, error) {
+	if err := exec.CommandContext(ctx, "sudo", "date").Run(); err != nil {
+		switch err.(type) {
+		case *exec.ExitError:
+			return false, nil
+		default:
+			return false, errors.Wrap(err, "error executing sudo date")
+		}
+	}
+
+	return true, nil
+}
+
+func dmesgContainsOOMKill(line string) bool {
+	return strings.Contains(line, "Out of memory") ||
+		strings.Contains(line, "Killed process") || strings.Contains(line, "oom")
```

Comment:
I'm concerned about just searching for "oom", since it potentially could be part of other words. Are there linux systems that print "oom" and not "Out of memory" or "Killed process"?

Reply:
I'm changing this to search "OOM killer" or "OOM-killer" as more specifically given in the original grep